### DISCUSSION TITLE
How to use "resolver" on useForm together with "rules" on each field?

### DISCUSSION THREAD

**Discussion by serhiiminin** [react-hook-form/react-hook-form#8992] [Q&A] [answered]
Hi all,
I am wrapping 3d party library controls with react-hook-form API ("useController"). 
What is the best practice for validation? I use yup on the level of form.
```
interface ComponentFormValues {
    title: string;
}

...
// My Form component
const methods = useForm<ComponentFormValues>({
   defaultValues,
   resolver: yupResolver(validationSchema),
});

<FormProvider {...methods}>{chidlren}</FormProvider>

...
// My InputField component

const {
   field: { onChange, onBlur, name, value, ref },
   fieldState: { error },
} = useController({ name: fieldName });

...
const validationSchema: yup.SchemaOf<ComponentFormValues> = yup
    .object({  title: yup.string().required('This field is required' )})
    .required();

<Form defaultValues={{title: "Hello"}} validationSchema={validationSchema}>
 <InputField/>
<Form/>

```

On the level of InputField, I want to know if a field is required. 
Is there a way to consume provided "yup" schema in useController inside InputField in order to determine if a fields is required (I want to render an asterisk there before the validation is done)? 
I want to have a single responsibility thing. I expect that if I provide validation schema on the Form level, I don't have to use "rules" on the level of InputField

**Comment by Moshyfawn** [comment, marked as answer]
There's no built-in way to do that. Take a look at #5258

**Reply by serhiiminin** [reply]
Thank you for the example. I was surprised that it was not described in the documentation. 
Overall, what is the point of having validation on both levels (Form and FormField)? 
Also, I am using FormProvider. It means that I need somehow consume my validation schema in the input field

**Reply by Moshyfawn** [reply]
You can't use both `rules` and `resolver` in the same form. If you opt-into using `resolver`, the field `rules` are ignored. Refer to [the schema rules section of RHF docs](https://react-hook-form.com/api/useform#schema-rules) for more info

**Reply by Moshyfawn** [reply]
> I was surprised that it was not described in the documentation.

What were you expecting to see in the docs?

**Reply by Moshyfawn** [reply]
> I am using FormProvider. It means that I need somehow consume my validation schema in the input field

If you define your validation schema inside the component, you can only pass it down as a prop or create a custom React Context to avoid prop drilling

**Reply by serhiiminin** [reply]
> What were you expecting to see in the docs?

Some information on how individual fields can consume validation schema from the provider. 
How would you approach it? Imagine, you provided validation schema with required fields to the resolver. How should a particular field know if it is required or not (without duplicating this information)?

**Reply by slava-ff** [reply]
same problem @serhiiminin 
have you resolved it somehow till now?
